src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/external_repacking_adjuster.cpp

```diff
@@ -12,35 +12,59 @@
 #include "transformations/snippets/x64/op/brgemm_cpu.hpp"
 #include "transformations/snippets/x64/op/brgemm_utils.hpp"
 
-namespace ov::intel_cpu {
+namespace ov::intel_cpu::pass {
 
 const size_t BrgemmExternalRepackingAdjuster::brgemm_kernel_rank = 2;
 
 BrgemmExternalRepackingAdjuster::BrgemmExternalRepackingAdjuster(const ov::snippets::lowered::LinearIRCPtr& linear_ir,
                                                                  const CPURuntimeConfigurator* configurator)
     : snippets::lowered::pass::RuntimeOptimizer(configurator) {
+    const auto& cpu_config = ov::as_type_ptr<CPURuntimeConfig>(m_configurator->get_config());
+    const auto& repacked_inputs_config = cpu_config->repacked_input_config;
     const auto& params = linear_ir->get_parameters();
-    for (size_t i = 0; i < params.size(); ++i) {
-        const auto& param = params[i];
-        const auto& shape_infer_consumers = ov::snippets::utils::get_first_child_shape_infer_expr_seq(param);
-        const auto& out = shape_infer_consumers.empty() ? param->get_output_port(0)
-                                                        : shape_infer_consumers.back()->get_output_port(0);
-        const auto consumers = out.get_connected_ports();
-
-        for (const auto& consumer : consumers) {
-            auto brgemm = ov::as_type_ptr<ov::intel_cpu::BrgemmCPU>(consumer.get_expr()->get_node());
-            if (brgemm && brgemm_utils::with_repacking(brgemm->get_type()) && consumer.get_index() == 1) {
-                const auto src_prc = brgemm->get_input_element_type(0);
-                const auto wei_prc = brgemm->get_input_element_type(1);
-                const auto isa = brgemm_utils::get_primitive_isa(src_prc, brgemm_utils::with_amx(brgemm->get_type()));
-                const auto inner_n_block = brgemm_utils::repacking::compute_inner_n_block(wei_prc);
-                const auto is_transposed_b =
-                    BrgemmCopyB::is_transposed(m_configurator->get_io_descs()[i]->get_layout());
-                auto config = BrgemmCopyBKernelConfig(src_prc, wei_prc, isa, false, is_transposed_b, inner_n_block);
-                m_executors[i] = std::make_shared<BrgemmCopyBKernelExecutor>(configurator->get_cache(), config);
+    for (const auto& [idx, _] : repacked_inputs_config) {
+        OPENVINO_ASSERT(idx < params.size(), "Incorrect index of repacked input");
+
+        m_executors[idx] = create_executor(params[idx], configurator->get_cache());
+        OPENVINO_ASSERT(m_executors[idx], "Repacked input executor must be inited!");
```

Comment:
Maybe it is worth to move this assert to `create_executor`? (I mean `OPENVINO_THROW` instead of `return nullptr;`) In this case, we don't need to check the result after each `create_executor` usage